kml/types.cpp

```diff
@@ -30,6 +30,42 @@ void SetBookmarksMinZoom(FileData & fileData, double countPerTile, int maxZoom)
   minZoomQuadtree.SetMinZoom(countPerTile, maxZoom, setMinZoom);
 }
 
+bool MultiGeometry::IsValid() const
+{
+  if (m_lines.empty())
+    return false;
+  if (!HasTimestamps())
+    return true;
+  for (size_t i = 0; i < m_lines.size(); ++i)
+  {
+    if (HasTimestampsFor(i) && m_lines[i].size() != m_timestamps[i].size())
+      return false;
+  }
+  return true;
+}
+
+void MultiGeometry::Clear() {
+  m_lines.clear();
+  m_timestamps.clear();
+}
+
+bool MultiGeometry::HasTimestamps() const
+{
+  if (m_timestamps.empty())
+    return false;
+  for (auto const & timestamp : m_timestamps)
+  {
+    if (!timestamp.empty())
+      return true;
```

Comment:
Does it imply that there can be mixed lines, with and without timestamps? Is it a valid condition?

Reply:
If the user mix the tracks manually this can be valid condition

Reply:
It would be great to clearly specify it in comments and related places. If I understood correctly, empty timestamp arrays will correspond to lines without timestamps (serialised into LineString). And non-empty ones - with timestamps (serialized to gx:MultiTrack). Correct?

If yes, then gx:MultiTrack should be used only if there are more than one line with timestamps.